Assets/LeapMotion/Scripts/DataStructures/Editor/Tests/ProduceConsumeBufferTest.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Threading;
+using NUnit.Framework;
+
+namespace Leap.Unity.Tests {
+
+  public class ProduceConsumeBufferTest {
+
+    private ProduceConsumeBuffer<TestStruct> buffer;
+
+    [SetUp]
+    public void Setup() {
+      buffer = new ProduceConsumeBuffer<TestStruct>(16);
+    }
+
+    [TearDown]
+    public void Teardown() {
+      buffer = null;
+    }
+
+    [Test]
+    [Timeout(1000)]
+    public void Test() {
+      Thread consumer = new Thread(new ThreadStart(consumerThread));
+      Thread producer = new Thread(new ThreadStart(producerThread));
+
+      consumer.Start();
+      producer.Start();
+
+      consumer.Join();
+      producer.Join();
+    }
+
+    private void consumerThread() {
```

Comment:
If a consumer or producer stalls, it is the same as that thread deciding to discontinue writing or reading, which is a safe and legitimate thing for a thread to do.  If a producer stalls, for example, the only thing that will happen is that no more data will be written into the buffer.  It is the consumers responsibility in this case to not get themselves stuck in an infinite loop waiting for data that may never arrive.